poker-app/components/PracticeButtonController.js

```diff
@@ -10,60 +10,92 @@ export default class PracticeButtonController extends Component {
             raises: 0,
             tag: "",
             tags: [],
-            gamesArray : []
+            gamesArray: []
         };
     };
 
-    populateGames(){
-        storageController.retrieveData().then((res)=> {
+    async populateGames() {
+        await storageController.retrieveData().then((res) => {
             let pastGames = JSON.parse(res);
-            this.setState({
-                gamesArray: pastGames
+            console.log("populate function");
+            let arrayOfgames = [];
```

Comment:
why do we need a new array?

Reply:
here i though that I should copy what is in storage to a local array and assign it to our state, since I need to loop through whats in storage and put it in a local array then assign that to our state.